src/ipc-handlers.ts

```diff
@@ -17,12 +17,15 @@ import nodePath from 'path';
 import * as Sentry from '@sentry/electron/main';
 import { __, LocaleData, defaultI18n } from '@wordpress/i18n';
 import archiver from 'archiver';
+import { z } from 'zod';
+import { CreateLoggerAction } from 'cli/commands/preview/logger-actions';
```

Comment:
If you're important code from the "cli" app, why not import the command and run it as a function instead of a subprocess. I don't have a fully formed opinion at the moment but I think right now we're doing a bit of both which is kind of weird.

Reply:
I suggested this approach of keeping the business logic squarely in the CLI when planning the project. It definitely adds complexity, but the upside is that it lets us avoid double bundling of PHP-WASM dependencies (once we get to moving that functionality from Studio to the CLI), which would increase the bundle size by ~700MB in the short term and ~400MB in the long term.

Reply:
As for code sharing, the idea is to only share modules without side effects. @bcotrim [suggested](https://github.com/Automattic/studio/pull/1201#discussion_r2044500053) moving those to a `common` folder for added clarity, which I think is a good idea.

Reply:
Sounds good to me ;)